### ISSUE TITLE
Zeronet waits an infinite time

### ISSUE THREAD

**Issue by sandrokensan** [HelloZeroNet/ZeroNet#225]
At start (python zeronet.py) when is executed FileServer.py in NAT Internet connection (port 43110 closed) It is shows:

```
python zeronet.py
- Starting ZeroNet...
[22:58:27] - OpenSSL loaded, version: 01000110F
[22:58:27] - Version: 0.3.4 r664, Python 2.7.9 (default, Dec 13 2014, 22:06:10) 
[GCC 4.8.2], Gevent: 1.0.2
[22:58:27] - Creating FileServer....
[22:58:27] - Creating UiServer....
[22:58:27] - Removing old SSL certs...
[22:58:27] - Starting servers....
[22:58:27] Ui.UiServer --------------------------------------
[22:58:27] Ui.UiServer Web interface: http://127.0.0.1:43110/
[22:58:27] Ui.UiServer --------------------------------------
[22:58:28] FileServer Checking port 15441 using portchecker.co...
[22:58:30] FileServer [BAD :(] Port closed: Port 15441 is closed.
[22:58:30] FileServer Checking port 15441 using canyouseeme.org...
```

and it waits an infinite time.

Log shows (Last rows):

```
[2015-12-13 23:04:18,537] DEBUG    Site:1Gfey7..fcdp Found 30 peers, new: 2
[2015-12-13 23:04:18,909] DEBUG    Site:1EU1tb..E4Vr Small number of peers detected...query all of peers using pex
[2015-12-13 23:04:18,910] DEBUG    Site:1EU1tb..E4Vr 152.236.101.192:15441 Getting connection...
[2015-12-13 23:04:18,910] DEBUG    FileServer Conn# 1 152.236.101.192 [?] > Connecting...
[2015-12-13 23:04:19,058] DEBUG    Site:1Gfey7..fcdp Announced port 0 to 6 trackers in 1.148s, errors: [], slow: []
[2015-12-13 23:04:19,058] DEBUG    Site:1Gfey7..fcdp Need connections: 3, Current: 0, Total: 29
[2015-12-13 23:04:19,058] DEBUG    Site:1Gfey7..fcdp 77.129.41.159:15441 Getting connection...
[2015-12-13 23:04:19,058] DEBUG    FileServer Conn# 2 77.129.41.159 [?] > Connecting...
[2015-12-13 23:04:19,337] DEBUG    FileServer Conn# 2 77.129.41.159 [v2] > Crypt out connection using: tls-rsa (server side: False)...
```

**Comment by HelloZeroNet** [comment]
Sounds like something with ssl connection, can you please specify your platform and try to start it with `zeronet.py --disable_encryption` ?

**Comment by sandrokensan** [comment]
Tanx a Lot! It is ok but is encryption needed?

```
python zeronet.py --disable_encryption
- Starting ZeroNet...
[23:23:17] - OpenSSL loaded, version: 01000110F
[23:23:17] - Version: 0.3.4 r664, Python 2.7.9 (default, Dec 13 2014, 22:06:10) 
[GCC 4.8.2], Gevent: 1.0.2
[23:23:17] - Creating FileServer....
[23:23:17] - Creating UiServer....
[23:23:17] - Removing old SSL certs...
[23:23:17] - Starting servers....
[23:23:17] Ui.UiServer --------------------------------------
[23:23:17] Ui.UiServer Web interface: http://127.0.0.1:43110/
[23:23:17] Ui.UiServer --------------------------------------
[23:23:18] FileServer Checking port 15441 using portchecker.co...
[23:23:20] FileServer [BAD :(] Port closed: Port 15441 is closed.
[23:23:20] FileServer Checking port 15441 using canyouseeme.org...
[23:23:28] FileServer [BAD :(] Port closed: Error: I could not see your service on 151.47.48.210 on port (15441) Reason: Connection timed out
[23:23:28] FileServer Trying to open port using UpnpPunch...
[23:23:44] FileServer Checking port 15441 using portchecker.co...
[23:23:46] FileServer [BAD :(] Port closed: Port 15441 is closed.
[23:23:46] FileServer Checking port 15441 using canyouseeme.org...
[23:23:54] FileServer [BAD :(] Port closed: Error: I could not see your service on 151.47.48.210 on port (15441) Reason: Connection timed out
[23:23:54] FileServer Upnp mapping failed :( Please forward port 15441 on your router to your ipaddress
[23:24:05] Site:1TaLkF..jipT data/users/1BwGNVyjmbqbuofH5WbwHgPUBtxZBvLQ7K/data.json file size does not match 1407 <> 1362, Hash: False
[23:24:05] Site:1TaLkF..jipT data/users/1BwGNVyjmbqbuofH5WbwHgPUBtxZBvLQ7K/data.json file size does not match 1407 <> 1362, Hash: False
...
```

P.S. Can I copy and past the old DATA dir in the new version of zeronet?

**Comment by HelloZeroNet** [comment]
Encryption is not needed, but recommended. Yes, you can copy old data dir to new.

**Comment by sandrokensan** [comment]
Is encription problem mine or is a bug?

**Comment by HelloZeroNet** [comment]
I had no similar report before. It's hard to tell whats the problem without the ability to reproduce the problem. Can you please specify your OS version?

**Comment by sandrokensan** [comment]
Yes, excuse me. I have linux, my distro is mageia 4, my kernel is $ uname -r
3.14.43-server-1.mga4

**Comment by sandrokensan** [comment]
It is a KDE.

**Comment by HelloZeroNet** [comment]
It was related to gevent 1.0.2, fixed in 44a68104fb867f193433aea2be0c791475ff3913
